service/route_handlers.go

```
@@ -554,25 +635,83 @@ func HandleSamlAssertion(w http.ResponseWriter, r *http.Request, assertion *saml
 		Path:   "/",
 	}
 	http.SetCookie(w, tokenCookie)
-
-	log.Debugf("redirecting the user with token to %v", redirectURL)
+	log.Debugf("redirecting the user with cookie %v to %v", tokenCookie, redirectURL)
 
 	http.Redirect(w, r, redirectURL, http.StatusFound)
 	return
 }
 
+func PostSamlTokenHTML(w http.ResponseWriter, r *http.Request) {
+	var token, finalRedirectURL string
+	var secure bool
+	var err error
+
+	r.ParseForm()
```

i think youre swallowing an error here